### PR TITLE
TEST: mpi/barrier test fix

### PR DESCRIPTION
**Pull request by vspetrov**
init/finalize reqs for multiple barriers

## What
Fixes incorrect UCC API usage in barrier test

### CODE REVIEW

File: test/mpi/test_barrier.cc [hunk 1]
```diff
@@ -77,6 +77,10 @@ void TestBarrier::run()
         if  (0 == rank && !completed) {
             MPI_Wait(&rreq, MPI_STATUS_IGNORE);
         }
+        if (i < size - 1) {
```
Comment (bureddy) [inline test/mpi/test_barrier.cc:80]: why not for the last rank?

Comment (vspetrov) [reply test/mpi/test_barrier.cc:80]: if this is the last iteration (i == size - 1) then i don't need to re-init for next iteration - there is no next iteration. coll_finalize is called in destructor.

**Comment by vspetrov** [comment]
@Sergei-Lebedev @bureddy added the coll_post/finalize_status_check

File: test/gtest/core/test_allgather.cc [hunk 2]
```diff
@@ -151,7 +155,7 @@ UCC_TEST_P(test_allgather_0, single)
     set_inplace(inplace);
     set_mem_type(mem_type);
 
-    data_init(size, dtype, count, ctxs);
+    data_init(size, dtype, count, ctxs, 0);
```
Comment (Sergei-Lebedev) [inline test/gtest/core/test_allgather.cc:158]: minor, parameter is bool so it's better to use true/false

File: src/core/ucc_coll.c [hunk 3]
```diff
@@ -196,17 +196,36 @@ UCC_CORE_PROFILE_FUNC(ucc_status_t, ucc_collective_init,
         ucc_coll_str(task, coll_debug_str, sizeof(coll_debug_str));
         ucc_debug("coll_init: %s", coll_debug_str);
     }
+    ucc_assert(task->super.status == UCC_OPERATION_INITIALIZED);
     *request = &task->super;
     return UCC_OK;
 }
 
+/* Check if yser is trying to post the request which is either in completed,
```
Comment (Sergei-Lebedev) [inline src/core/ucc_coll.c:204]: yser -> user

**Review by Sergei-Lebedev — APPROVED** [review]

**Review by bureddy — APPROVED** [review]

**Comment by vspetrov** [comment]
bot:retest

**Comment by vspetrov** [comment]
bot:retest

**Pull request merged**